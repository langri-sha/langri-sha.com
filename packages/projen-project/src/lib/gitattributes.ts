import {
  GitAttributesFile as BaseGitAttributesFile,
  type IResolver,
} from 'projen'

export class GitAttributesFile extends BaseGitAttributesFile {
  protected override synthesizeContent(
    resolver: IResolver,
  ): string | undefined {
    const content = super.synthesizeContent(resolver)?.split('\n')

    if (!content) {
      return
    }

    const [marker, blank, ...rest] = content
    rest.sort((a, b) => a.localeCompare(b))

    return [marker, blank, ...rest, ''].join('\n')
  }
}
