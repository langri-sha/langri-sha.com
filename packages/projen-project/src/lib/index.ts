export * from './node-package.js'
export * from './projenrc.js'
