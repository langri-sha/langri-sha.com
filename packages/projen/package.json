{
  "name": "@langri-sha/projen",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*",
    "projen": "0.81.1"
  },
  "peerDependencies": {
    "projen": "^0.81.1",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
