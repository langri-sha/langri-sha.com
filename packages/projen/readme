# @langri-sha/projen

Collection of [projen] templates for bootstraping monorepos and workspace projects.

[projen]: https://projen.dev/
