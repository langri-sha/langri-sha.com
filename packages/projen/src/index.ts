import {
  Project as BaseProject,
  type ProjectOptions as BaseProjectOptions,
} from 'projen'

export interface ProjectOptions extends BaseProjectOptions {}

export class Project extends BaseProject {
  constructor(options: ProjectOptions) {
    super({
      ...options,
      gitIgnoreOptions: getGitIgnoreOptions(options),
    })
  }
}

const getGitIgnoreOptions = (
  options: ProjectOptions,
): ProjectOptions['gitIgnoreOptions'] => ({
  ...options.gitIgnoreOptions,
  ignorePatterns: [
    ...`
    .*
    !.babelrc
    !.dockerignore
    !.editorconfig
    !.eslintignore
    !.eslintrc.js
    !.gitattributes
    !.gitignore
    !.gitkeep
    !.husky
    !.npmignore
    !.openssl
    !.prettierignore
    !.terraform.lock.hcl
    *.log
    *.tsbuildinfo

    !.github/
    !.husky/
    !.projen/
    dist/
    local/
    node_modules/
    `
      .split('\n')
      .map((pattern) => pattern.trim())
      .filter((pattern) => pattern.length > 0),
    ...(options.gitIgnoreOptions?.ignorePatterns ?? []),
  ],
})
