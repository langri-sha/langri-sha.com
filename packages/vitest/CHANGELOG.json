{
  "name": "@langri-sha/vitest",
  "entries": [
    {
      "date": "Sat, 27 Jul 2024 15:50:17 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d542415a6b2c2c339a8367c77b925842000f9f9a",
            "comment": "feat(workspace): Exclude compiling test modules"
          }
        ]
      }
    },
    {
      "date": "Fri, 26 Jul 2024 21:00:29 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(workspace): Migrate documentation pathnames"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(projen): Add Git ignore patterns"
          }
        ]
      }
    },
    {
      "date": "Wed, 17 Jul 2024 10:48:15 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "chore(deps): chore(deps): update dependency vitest to v2"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "fix(workspace): Use correct GitHub directory"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "feat(root): Migrate to ESM"
          }
        ]
      }
    },
    {
      "date": "Mon, 08 Jul 2024 21:43:05 GMT",
      "version": "0.1.0",
      "tag": "@langri-sha/vitest_v0.1.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Use `vitest`"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(vitest): Add `nock`"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Add repository path"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(tsconfig): Use `${configDir}`"
          }
        ],
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use correct output directory"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use workspace `@langri-sha/tsconfig`"
          }
        ]
      }
    }
  ]
}
