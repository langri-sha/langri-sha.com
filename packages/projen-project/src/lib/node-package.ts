import * as fs from 'node:fs'

import { Project, javascript } from 'projen'

/**
 * Options for configuring Node.js packages.
 */
export interface NodePackageOptions extends javascript.NodePackageOptions {
  peerDependenciesMeta?: Record<string, { optional?: boolean }>
  type?: 'commonjs' | 'module'
}

/**
 * Custom Node.js package manager that preserves package.
 */
export class NodePackage extends javascript.NodePackage {
  constructor(project: Project, options?: NodePackageOptions) {
    super(project, options)

    this.addVersion(this.#getPackageJson()?.version ?? '0.0.0')

    if (options?.type) {
      this.addField('type', options.type)
    }

    if (options?.peerDependenciesMeta) {
      this.addField('peerDependenciesMeta', options.peerDependenciesMeta)
    }

    // Reset, so we can run `pnpx projen` on CI after Renovate upgrades.
    this.project.tasks
      .tryFind('install:ci')
      ?.reset(this.project.tasks.tryFind('install')?.steps[0]?.exec)
  }

  #getPackageJson() {
    try {
      return JSON.parse(fs.readFileSync(this.file.absolutePath, 'utf-8'))
    } catch (_) {
      return undefined
    }
  }

  override postSynthesize(): void {
    if (this.project.parent) {
      return
    }

    super.postSynthesize()
  }
}
