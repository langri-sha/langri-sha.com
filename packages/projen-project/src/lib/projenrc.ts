import { ProjenrcFile as BaseProjenrcFile, Project, SampleFile } from 'projen'

export interface ProjenrcOptions {
  /**
   * The name of the projenrc file.
   * @default ".projenrc.mts"
   */
  readonly filename?: string
}

export class ProjenrcFile extends BaseProjenrcFile {
  public override readonly filePath: string

  constructor(
    project: Project,
    { filename = '.projenrc.mts' }: ProjenrcOptions,
  ) {
    super(project)

    this.filePath = filename
    this.project.defaultTask?.exec(
      `node --loader ts-node/esm/transpile-only ${filename}`,
    )

    new SampleFile(project, filename, {
      contents: `import { Project } from '@langri-sha/projen-project'

      const project = new Project()

      project.synth()
      `,
    })
  }
}
