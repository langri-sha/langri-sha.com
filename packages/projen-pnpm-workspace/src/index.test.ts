import { synthSnapshot } from 'projen/lib/util/synth'
import { expect, test } from '@langri-sha/vitest'

import { Project } from 'projen'
import { PnpmWorkspace } from './index'
import { NodePackage } from 'projen/lib/javascript'

test('defaults', () => {
  const project = new Project({
    name: 'test-project',
  })

  new PnpmWorkspace(project)

  expect(synthSnapshot(project)['pnpm-workspace.yaml']).toMatchSnapshot()
})

test('with filename', () => {
  const project = new Project({
    name: 'test-project',
  })

  new PnpmWorkspace(project, { filename: 'pnpm-workspace.yml' })

  expect(synthSnapshot(project)['pnpm-workspace.yml']).toMatchSnapshot()
})

test('with packages', () => {
  const project = new Project({
    name: 'test-project',
  })

  new PnpmWorkspace(project, {
    packages: ['apple', 'banana'],
  })

  expect(synthSnapshot(project)['pnpm-workspace.yaml']).toMatchSnapshot()
})

test('sorts packages', () => {
  const project = new Project({
    name: 'test-project',
  })

  new PnpmWorkspace(project, {
    packages: ['b', 'a'],
  })

  expect(synthSnapshot(project)['pnpm-workspace.yaml']).toMatchSnapshot()
})

test('dedupes packages', () => {
  const project = new Project({
    name: 'test-project',
  })

  new PnpmWorkspace(project, {
    packages: ['a', 'a'],
  })

  expect(synthSnapshot(project)['pnpm-workspace.yaml']).toMatchSnapshot()
})

test('adds packages from subproject NodePackages', () => {
  const project = new Project({
    name: 'test-project',
  })

  new PnpmWorkspace(project)

  const subprojectA = new Project({
    name: 'subproject-a',
    outdir: 'subprojects/subproject-a',
    parent: project,
  })

  const subprojectB = new Project({
    name: 'subproject-b',
    outdir: 'subprojects/subproject-b',
    parent: project,
  })

  const subsubproject = new Project({
    name: 'subproject-b-subproject',
    outdir: 'subprojects/subproject-b/subproject',
    parent: project,
  })

  const standalone = new Project({
    name: 'standalone',
    outdir: 'some/where/standalone',
    parent: project,
  })

  new NodePackage(project)
  new NodePackage(subprojectA)
  new NodePackage(subprojectB)
  new NodePackage(subsubproject)
  new NodePackage(standalone)

  expect(synthSnapshot(project)['pnpm-workspace.yaml']).toMatchSnapshot()
})
