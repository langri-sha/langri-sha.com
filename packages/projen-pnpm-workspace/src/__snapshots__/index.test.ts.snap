// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds packages from subproject NodePackages 1`] = `
"# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".

packages:
  - some/where/standalone
  - subprojects/*
  - subprojects/subproject-b/subproject
"
`;

exports[`dedupes packages 1`] = `
"# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".

packages:
  - a
"
`;

exports[`defaults 1`] = `
"# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".

packages: []
"
`;

exports[`sorts packages 1`] = `
"# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".

packages:
  - a
  - b
"
`;

exports[`with filename 1`] = `
"# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".

packages: []
"
`;

exports[`with packages 1`] = `
"# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".

packages:
  - apple
  - banana
"
`;
